### PR TITLE
Update prettier, eslint and plugins to latest

### PR DESCRIPTION
**Pull request by mjs**
- Updated eslint+prettier config as per https://alligator.io/vuejs/vue-eslint-prettier/
- Reformatted everything according to new setup
- Fix problems detected by newer eslint version 
- Document eslint and prettier requirements

### CODE REVIEW

File: src/components/Devices/DeviceListing.vue [hunk 1]
```diff
@@ -1,40 +1,40 @@
...
+          :position="'right'"
+        />
       </router-link>
     </div>
   </div>
 </template>
 
 <script>
 import { mapState } from "vuex";
-import DeviceDetail from "./DeviceDetail.vue";
```
Comment (gferraro) [inline src/components/Devices/DeviceListing.vue]: Did prettier remove this automatically?

File: src/views/RegisterView.vue [hunk 2]
```diff
@@ -123,16 +124,19 @@ export default {
         }
         return `Minimum username length is ${usernameLength} characters`;
       }
+      return null;
```
Comment (gferraro) [inline src/views/RegisterView.vue]: did it also add these?

File: src/components/Devices/DeviceListing.vue (hunk 1 above)
Comment (mjs) [reply src/components/Devices/DeviceListing.vue]: No this was me, based on warnings from the new version of eslint.

File: src/views/RegisterView.vue (hunk 2 above)
Comment (mjs) [reply src/views/RegisterView.vue]: Again this was me. The new eslint version picks up more stuff.

**Pull request merged**